core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -119,6 +119,14 @@ object TopicCommand extends Logging {
         AdminUtils.addPartitions(zkClient, topic, nPartitions, replicaAssignmentStr)
         println("Adding partitions succeeded!")
       }
+      if(opts.options.has(opts.replicationFactorOpt)) {
+        val replicationFactor = opts.options.valueOf(opts.replicationFactorOpt)
+        val brokerList = ZkUtils.getSortedBrokerList(zkClient)
+        val partitions = ZkUtils.getPartitionsForTopics(zkClient, Seq(topic))(topic)
+        val replicaAssignment = AdminUtils.assignReplicasToBrokers(brokerList, partitions.length, replicationFactor)
```

Comment:
This will cause the partition to be randomly redistributed and might cause unnecessary data move. Maybe we should specify the fixedStartIndex.